examples/example-generation/translate.py

```diff
@@ -0,0 +1,303 @@
+            template = Template(self.targetDict["Expr"]["Enum"])
+            return template.substitute(type=expr[key][0],value=expr[key][1]["Identifier"])
+
+        raise Exception("Unknown expression type: " + key)
+
+    def translatePrint(self, printStmt):
+        template = Template(self.targetDict["Print"])
+        return template.substitute(expr=self.translateExpr(printStmt["Expr"]))
+
+    def translateComment(self, commentString):
+        template = Template(self.targetDict["Comment"])
+        return template.substitute(comment=commentString)
+
+    def translateType(self, type):
+        """ Translate type AST
+        Args:
+            type: object like {"ObjectType": "IntMatrix"}, {"BasicType": "float"}, etc.
+        """
+        template = ""
+        typeKey = type.keys()[0]
+
+        # Store dependency
+        if typeKey == "ObjectType":
+            self.dependencies["AllClasses"].add(type[typeKey])
+
+        if type[typeKey] in self.targetDict["Type"]:
+            template = Template(self.targetDict["Type"][type[typeKey]])
+        else:
+            template = Template(self.targetDict["Type"]["Default"])
+
+        return template.substitute(type=type[typeKey])
+
+    def translateArgumentList(self, argumentList):
+        """ Translate argument list AST
+        Args:
+            argumentList: object like None, {"Expr": exprAST},
+                          [{"Expr": exprAST}, {"Expr": exprAST}], etc.
+        """
+        if argumentList == None or argumentList == []:
+            return ""
+        if isinstance(argumentList, list):
+            head = argumentList[0]
+            tail = argumentList[1:]
+
+            translation = self.translateArgumentList(head)
+            if len(tail) > 0:
+                translation += ", " + self.translateArgumentList(tail)
+
+            return translation
+        else:
+            if "Expr" in argumentList:
+                return self.translateExpr(argumentList["Expr"])
+            elif "ArgumentList" in argumentList:
+                return self.translateArgumentList(argumentList["ArgumentList"])
+
+def translate(ast, targetDict):
+    translator = Translator(targetDict)
+    programName = os.path.basename(ast["FilePath"]).split(".")[0]
+    return translator.translateProgram(ast["Program"], programName)
+
+def loadTargetDict(targetJsonPath):
+    try:
+        with open(targetJsonPath, "r") as jsonDictFile:
+            return json.load(jsonDictFile)
+    except IOError, err:
+        if err.errno == 2:
+            print "Target \"" + targetJsonPath + "\" does not exist"
+        else:
+            print err
+        raise Exception("Could not load target dictionary")
+
+if __name__ == "__main__":
+    # Extract target language from arguments (default target is python)
+    targets = [target for target in map(lambda arg:arg[len("--target="):] if "--target=" in arg else "", sys.argv) if target != ""]
```

Comment:
Sorry I get back about that a bit late

There is a better standard api to do that: https://docs.python.org/2/library/argparse.html. Could you please change the code to use that so it gets much more flexible and easier to read?